var/spack/repos/builtin/packages/bigdft-core/package.py

```diff
@@ -67,33 +67,33 @@ def configure_args(self):
         linalg.append(spec["blas"].libs.ld_flags)
 
         args = [
-            "FCFLAGS=%s" % " ".join(openmp_flag),
-            "--with-ext-linalg=%s" % " ".join(linalg),
-            "--with-pyyaml-path=%s" % pyyaml,
-            "--with-futile-libs=%s" % spec["bigdft-futile"].libs.ld_flags,
-            "--with-futile-incs=%s" % spec["bigdft-futile"].headers.include_flags,
-            "--with-chess-libs=%s" % spec["bigdft-chess"].libs.ld_flags,
-            "--with-chess-incs=%s" % spec["bigdft-chess"].headers.include_flags,
-            "--with-psolver-libs=%s" % spec["bigdft-psolver"].libs.ld_flags,
-            "--with-psolver-incs=%s" % spec["bigdft-psolver"].headers.include_flags,
-            "--with-libABINIT-libs=%s" % spec["bigdft-libabinit"].libs.ld_flags,
-            "--with-libABINIT-incs=%s" % spec["bigdft-libabinit"].headers.include_flags,
-            "--with-libgain-libs=%s" % spec["libgain"].libs.ld_flags,
-            "--with-libgain-incs=%s" % spec["libgain"].headers.include_flags,
+            f"FCFLAGS={' '.join(openmp_flag)}",
+            f"--with-ext-linalg={' '.join(linalg)}",
+            f"--with-pyyaml-path={pyyaml}",
+            f"--with-futile-libs={spec['bigdft-futile'].libs.ld_flags}",
+            f"--with-futile-incs={spec['bigdft-futile'].headers.include_flags}",
+            f"--with-chess-libs={spec['bigdft-chess'].libs.ld_flags}",
+            f"--with-chess-incs={spec['bigdft-chess'].headers.include_flags}",
+            f"--with-psolver-libs={spec['bigdft-psolver'].libs.ld_flags}",
+            f"--with-psolver-incs={spec['bigdft-psolver'].headers.include_flags}",
+            f"--with-libABINIT-libs={spec['bigdft-libabinit'].libs.ld_flags}",
+            f"--with-libABINIT-incs={spec['bigdft-libabinit'].headers.include_flags}",
+            f"--with-libgain-libs={spec['libgain'].libs.ld_flags}",
+            f"--with-libgain-incs={spec['libgain'].headers.include_flags}",
             "--with-libxc-libs=%s %s"
             % (spec["libxc"].libs.ld_flags, spec["libxc"].libs.ld_flags + "f90"),
```

Comment:
^

Reply:
Are you suggesting to use `self.define()` here as well? How does that work for `--with` statements?